### DISCUSSION TITLE
Dwg2007 version exception

### DISCUSSION THREAD

**Discussion by yingyue4008** [LibreDWG/libredwg#642] [Q&A]
Excuse me, after reading the 2007 version of dwg, the error is 64,Dwg_Data writes to the new file error is 8192,2007 version How to read and write?

**Comment by rurban** [comment]
You can only read, writes for r2007 are not possible yet. you need to write in r2000 format. r2007 can read it then easily.

64 is DWG_ERR_VALUEOUTOFBOUNDS
8192 is DWG_ERR_OUTOFMEM, see include/dwg.h

**Reply by yingyue4008** [reply]
> You can only read, writes for r2007 are not possible yet. you need to write in r2000 format. r2007 can read it then easily.
> 
> 64 is DWG_ERR_VALUEOUTOFBOUNDS 8192 is DWG_ERR_OUTOFMEM, see include/dwg.h

I changed the version number to R2000 before writing, but I opened dwg and reported a fatal version error

error:Unhandled Access Violation Reading 0x0045 Exception at 430c3a85h

**Reply by yingyue4008** [reply]
> You can only read, writes for r2007 are not possible yet. you need to write in r2000 format. r2007 can read it then easily.
> 
> 64 is DWG_ERR_VALUEOUTOFBOUNDS 8192 is DWG_ERR_OUTOFMEM, see include/dwg.h

Excuse me, How do I set it to r2000？

**Reply by rurban** [reply]
set dwg->header.version to R_2000.
via the API this is set automatically

**Comment by iisaxin** [comment]
你可以先把高版本文件另存为2000版本的再进行读取可以吗？
